GameOfRunes/GameOfRunes/Models/Stage.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  Stage.swift
+//  GameOfRunes
+//
+//  Created by Jermy on 29/3/20.
+//  Copyright © 2020 TeamHoWan. All rights reserved.
+//
+
+import UIKit
+
+struct Stage {
+    enum Category: String {
+        case normal
+        case boss
+    }
+    enum AchievementLevel: String {
+        case S
+        case A
+        case B
+        case C
+        case empty
+    }
+    
+    let name: String
+    let groupName: String
+    let category: Category
+    let relativePositionRatioInMap: (x: CGFloat, y: CGFloat)
+    let arena: ArenaType
+    let difficulty: Int
+    let numWaves: Int
+    // might want to add the array of enemies here? But how do we determine the numWaves?
```

Comment:
num waves should be hardcoded i guess, we can add the array of enemies here, and persist this entire class?

Reply:
Yep, that's the plan

Reply:
So the `numWaves` value here will dictate the spawn engine's behaviour?